### PR TITLE
Added --delete-dir-on-start option for babel 

### PR DESCRIPTION
**Pull request by keshcode**
https://github.com/babel/babel/blob/7.0/CONTRIBUTING.md , added `--delete-dir-on-start` option to remove the deleted files or directory in the source directory from the out directory #5636

| Q | A 
| ------------------------ | ---
| Patch: Bug Fix? | No
| Major: Breaking Change? | No
| Minor: New Feature? | yes
| Deprecations? | No
| Spec Compliancy? | No
| Tests Added/Pass? | No
| Fixed Tickets | Fixes [#5636 ](https://github.com/babel/babel/issues/5636)
| License | MIT
| Doc PR | yes,  
| Dependency Changes | No

Added deleteDir function in the packages/babel-cli/src/babel/util.js that recursuvely delete all the files and directory in the directory how's path is passed .

Added option with `-l` or` --delete-dir-on-start`  and added checks that this option can only be used with `--out-dir` in packages/babel-cli/src/babel/index.js

called util.deleteDir function in  packages/babel-cli/src/babel/dir.js just before compilation to delete the the out directory

### CODE REVIEW

**Comment by keshcode** [comment]
can anyone help me with circle CI and Travis-ci/pr 
it worked fine on my pc

**Comment by nicolo-ribaudo** [comment]
I have never contributed to this part of babel, but I think you either need to add the new option to [this set](https://github.com/keshcode/babel/blob/3f24d492a7915b0c53882f3517d7c5535c4b83c3/packages/babel-core/src/config/option-manager.js#L24) -- if you think that it should be possible to enable this option from .babelrc -- or delete it [here](https://github.com/keshcode/babel/blob/3f24d492a7915b0c53882f3517d7c5535c4b83c3/packages/babel-cli/src/babel/index.js#L136) -- if you think this should be a cli-only option

**Comment by keshcode** [comment]
@nicolo-ribaudo  it's only a cli issue and have unit test on my end ,it would be great help if u could help me solve this cli issue

File: packages/babel-cli/src/babel/index.js [hunk 1]
```diff
@@ -74,6 +74,7 @@ commander.option("-o, --out-file [out]", "Compile all input files into a single
 commander.option("-d, --out-dir [out]", "Compile an input directory of modules into an output directory");
 commander.option("-D, --copy-files", "When compiling a directory copy over non-compilable files");
 commander.option("-q, --quiet", "Don't log anything");
+commander.option("-l, --delete-dir-on-start", "Delete's the out directory before compilation");
```
Comment (jridgewell) [inline packages/babel-cli/src/babel/index.js]: Why `-l`?

Comment (keshcode) [inline packages/babel-cli/src/babel/index.js]: is there a naming convention to be followed, I wasn't aware of that.  I have no idea why  I used  `-l`

Comment (jridgewell) [inline packages/babel-cli/src/babel/index.js]: I would probably drop the shorthand, since it isn't obvious.

Comment (existentialism) [inline packages/babel-cli/src/babel/index.js]: Also, maybe:

> Delete the directory specified in --out-dir before compilation

Comment (keshcode) [inline packages/babel-cli/src/babel/index.js]: @jridgewell I could drop the shorthand, but currently I am stuck with ci

Comment (existentialism) [inline packages/babel-cli/src/babel/index.js]: @keshcode re: CI failures, have you looked at https://github.com/babel/babel/blob/7.0/packages/babel-core/src/config/option-manager.js#L29?

**Comment by existentialism** [comment]
@keshcode re: CI failures, @nicolo-ribaudo's suggestion is correct.

**Pull request closed without merging**